src/main/java/com/example/demo/service/impl/BookServiceImpl.java

```diff
@@ -33,4 +34,19 @@ public BookDto getById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can not get book by id:" + id)));
     }
+
+    @Override
+    public BookDto update(CreateBookRequestDto createBookRequestDto, Long id) {
+        Book book = bookMapper.toModel(createBookRequestDto);
+        book.setId(id);
+        if (getById(id) != null) {
+            return bookMapper.toDto(bookRepository.save(book));
+        }
+        throw new EntityNotFoundException("Book with this id doesn't exist. Id:" + id);
```

Comment:
check should go at the first order
```suggestion
        if (getById(id) == null) {
            throw new EntityNotFoundException("Book with this id doesn't exist. Id:" + id);
        }
        Book book = bookMapper.toModel(createBookRequestDto);
        book.setId(id);
        return bookMapper.toDto(bookRepository.save(book));
```